src/DataProtection/Cryptography.Internal/src/CryptoUtil.cs

```diff
@@ -91,6 +91,21 @@ public static bool TimeConstantBuffersAreEqual(byte* bufA, byte* bufB, uint coun
 #endif
     }
 
+#if NET10_0_OR_GREATER
+    [MethodImpl(MethodImplOptions.NoInlining | MethodImplOptions.NoOptimization)]
+    public static bool TimeConstantBuffersAreEqual(ReadOnlySpan<byte> bufA, ReadOnlySpan<byte> bufB)
+    {
+        // Technically this is an early exit scenario, but it means that the caller did something bizarre.
+        // An error at the call site isn't usable for timing attacks.
```

Comment:
```suggestion
This early exit handles unexpected input without introducing timing vulnerabilities.
```

Reply:
can you please elaborate? I should not do `Assert()` here?

Reply:
> can you please elaborate? I should not do `Assert()` here?


Are you asking me? 
I have no objection to using `Assert`

I replaced your colloquial comments with  formalized/professional comments. 
If my changes are inaccurate, modify my suggestion. 

`Technically` adds no value. The MS style guide mantra is less is more. Bizarre is colloquial and it's ***NOT*** a verb used in computer science to describe an invalid state.

Reply:
@BrennanConroy  can you suggest better comments?  I might be misunderstanding @DeagleGross  code comments.

Reply:
I thought you are proposing something about the code itself.
Comments were just copied from the other method (.net fx implementation).

i will adjust them, thanks for bringing this in

Reply:
changed the wording